lib/Sema/TypeCheckSwitchStmt.cpp

```diff
@@ -973,6 +973,10 @@ namespace {
       }
     }
 
+    static bool hasStrictExhaustiveChecksEnabled(ASTContext &Context) {
+      return !(Context.Diags.isIgnoredDiagnostic(diag::decompose_default_case.ID));
```

Comment:
Nit: If you run `clang-format` does it add the parenthesis? They are not really necessary.